app/shared/telemetry/telemetry.py

```
@@ -0,0 +1,188 @@
+    def instrument_fastapi(self, app: FastAPI) -> None:
+        """Instrument FastAPI (requests, duration, status, exceptions)."""
+        if not self.enabled or not self.tracer_provider:
+            return
+        try:
+            FastAPIInstrumentor.instrument_app(
+                app,
+                tracer_provider=self.tracer_provider,
+                excluded_urls="/health,/metrics",
+            )
+            logger.info("FastAPI instrumentation enabled")
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)